frontend/src/metabase/visualizations/echarts/cartesian/model/axis.ts

```diff
@@ -471,14 +473,14 @@ export function getYAxesModels(
 
   return {
     leftAxisModel: getYAxisModel(
-      leftAxisSeries,
+      leftAxisSeries.map(dataKey => seriesByDataKey[dataKey]),
```

Comment:
Not sure about this — I also tried reworking `getYAxisSplit` to return series models instead of data keys, but seems like it'd take more refactoring. Please let me know if we should dig into that